docs/sso/migrate_v1_v2.md

```diff
@@ -0,0 +1,92 @@
+# Migrating from SSO v1 to v2
+
+This document provides a general outline of the steps required for migrating from SSO v1 to SSO v2. Not all steps mentioned here will be 100% applicable for all use cases but should be generic enough to adjust for a variety of situations.
+
+More details about the v1 deprecation can be found in [this blog post](https://developers.eveonline.com/blog/article/sso-endpoint-deprecations-2)
+
+## Outline
+
+1. Use form encoding for `/oauth/token` requests. JSON and Query String are no longer supported
+2. Always update your stored refresh token with the latest received from the `/oauth/token endpoint`
+3. Change the base url from `login.eveonline.com` to `login.eveonline.com/v2`
+4. Replace calls to `/oauth/verify` with local decode and optional verification
+
+The migration steps don't have to be implemented in this exact order, but doing so has some advantages, such as making an explicit token migration unnecessary. Old v1 refresh_tokens will simply be replaced with the new v2 token on the first refresh.
+
+----
+
+## 1. Use form encoding for token requests
+
+The SSO v1 endpoints supported requests using a number of different payload formats, including JSON and URL Query String. SSO v2 enforces the payload to be sent in the body as an url-encoded string, with the `Content-Type: application/x-www-form-urlencoded` header.
+
+In most modern HTTP clients, changing that should be trivial. For example, making the change in Python `requests` looks like this:
+
+```python
+payload = {"client_id": 12345678, grant_type: "authorization_code" [...]}
+
+# From this
+- requests.post("https://login.eveonline.com/oauth/token", json=payload)
+
+# To this
++ requests.post("https://login.eveonline.com/v2/oauth/token", data=payload)
```

Comment:
note that the refresh token which you receive from the v2/oauth/token endpoint **is not url-encoded**. 
this means it is not possible to just take the old token and put it into the payload instead you have to URL-encode it before. 

for example in v1 creating the form data for the oauth/token endpoint looked like this: 
`// V1 go code`
`body := fmt.Sprintf("grant_type=refresh_token&refresh_token=%s&client_id=%s", char.InitAuth.RefreshToken, clientID)`
`// V2 go code`
`URLEncodedToken := url.QueryEscape(char.InitAuth.RefreshToken)`
`body2 := fmt.Sprintf("grant_type=refresh_token&refresh_token=%s&client_id=%s",
			URLEncodedToken, clientID)`